shared/constants/chat2/meta.js

```diff
@@ -0,0 +1,325 @@
+// Meta manages the metadata about a conversation. Participants, isMuted, reset people, etc. Things that drive the inbox
+// @flow
+import * as I from 'immutable'
+import * as RPCChatTypes from '../types/rpc-chat-gen'
+import * as RPCTypes from '../types/rpc-gen'
+import * as Types from '../types/chat2'
+import type {_ConversationMeta} from '../types/chat2/meta'
+import {formatTimeForConversationList} from '../../util/timestamp'
+import {globalColors} from '../../styles'
+import {isMobile, isIOS, isAndroid} from '../platform'
+import {parseFolderNameToUsers} from '../../util/kbfs'
+import {toByteArray} from 'base64-js'
+
+const IGNORE_UNTRUSTED_CACHE = false
+IGNORE_UNTRUSTED_CACHE && console.log('aaa NOJIMA skipping locametadata')
+
+const conversationMemberStatusToMembershipType = (m: RPCChatTypes.ConversationMemberStatus) => {
+  switch (m) {
+    case RPCChatTypes.commonConversationMemberStatus.active:
+      return 'active'
+    case RPCChatTypes.commonConversationMemberStatus.reset:
+      return 'youAreReset'
+    default:
+      return 'youArePreviewing'
+  }
+}
+
+// This one call handles us getting a string or a buffer
+const supersededConversationIDToKey = (id: string | Buffer): string =>
+  typeof id === 'string' ? Buffer.from(toByteArray(id)).toString('hex') : id.toString('hex')
+
+export const unverifiedInboxUIItemToConversationMeta = (
+  i: RPCChatTypes.UnverifiedInboxUIItem,
+  username: string
+) => {
+  if (IGNORE_UNTRUSTED_CACHE) {
+    // $FlowIssue flow rightfully complains
+    i.localMetadata = null
+  }
+
+  // Private chats only
+  if (i.visibility !== RPCTypes.commonTLFVisibility.private) {
+    return null
+  }
+
+  // We only treat implied adhoc teams as having resetParticipants
+  const resetParticipants = I.Set(
+    i.localMetadata &&
+    (i.membersType === RPCChatTypes.commonConversationMembersType.impteamnative ||
+      i.membersType === RPCChatTypes.commonConversationMembersType.impteamupgrade) &&
+    i.localMetadata.resetParticipants
+      ? i.localMetadata.resetParticipants
+      : []
+  )
+
+  const participants = I.OrderedSet(
+    i.localMetadata
+      ? i.localMetadata.writerNames || []
+      : parseFolderNameToUsers(username, i.name).map(ul => ul.username)
+  )
+
+  const channelname =
+    i.membersType === RPCChatTypes.commonConversationMembersType.team && i.localMetadata
+      ? i.localMetadata.channelName
+      : ''
```

Comment:
Maybe something better thank blank? Like a dash or `(loading)`

Reply:
if the goal is to make it show up like that in the inbox then we should do that there and not in our data layer

Reply:
Sounds good, I think they just don't show up at all now.